js/adapt-stateful-session.js

```diff
@@ -79,6 +79,24 @@ define([
 			this.listenTo(Adapt.blocks, 'change:_isComplete', this.onBlockComplete);
 			this.listenTo(Adapt.course, 'change:_isComplete', this.onCompletion);
 			this.listenTo(Adapt, 'assessment:complete', this.onAssessmentComplete);
+			this.listenTo(Adapt, 'app:resetSession', this.onResetSession);
+		},
+
+		removeEventListeners: function () {
+			$(window).off('unload', this._onWindowUnload);
```

Comment:
onResetSession calls `this.setupEventListeners` this function calls `$(window).on('unload', this._onWindowUnload)`.